src/main/java/org/kohsuke/github/Requester.java

```diff
@@ -281,9 +283,9 @@ public Requester with(String key, Object value) {
      * @return the requester
      */
     public Requester set(String key, Object value) {
-        for (Entry e : args) {
-            if (e.key.equals(key)) {
-                e.value = value;
+        for (int index = 0; index < args.size(); index++) {
```

Comment:
With `final` Entry fields, we now replace the instance rather than changing the values.